### DISCUSSION TITLE
Update kernel to 6.1

### DISCUSSION THREAD

**Discussion by squassina** [7Ji/orangepi5-archlinuxarm#37] [Q&A] [answered]
Hello! I'm glad to see such a project as it's a new challenge to have a different distro in my Orange Pi, one that I never used.

I was able to install and configure according to the instructions, but I wonder if it's possible to update the kernel to the latest 6.1. If I download the sources, compile and install it would be enough? Do you have any plans to do that?

Thanks!

**Comment by 7Ji** [comment, marked as answer]
https://github.com/7Ji/archrepo/blob/1e8a1cdcc9bcdbcbcdf004ff2df7cf2c8d2ecbe9/aarch64.yaml#L26

6.1 kernel is available in my repo

Just `pacman -Syu linux-aarch64-rockchip-bsp6.1-joshua-git`

**Reply by squassina** [reply]
Thanks! I didn't know that!

**Reply by squassina** [reply]
It's returning 404:

```
pacman -Syu linux-aarch64-rockchip-bsp6.1-joshua-git
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 alarm is up to date
 aur is up to date
 7Ji                                          15.0 KiB  11.1 KiB/s 00:01 [########################################] 100%
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (1) linux-aarch64-rockchip-bsp6.1-joshua-git-6.1.43.r1266115.gc2daa578-1

Total Download Size:    41.84 MiB
Total Installed Size:  200.38 MiB

:: Proceed with installation? [Y/n] 
:: Retrieving packages...
 linux-aarch64-rockchip-bsp6.1-joshua-git-6.1.43.r1266115.gc2daa578-1-aarch64.pkg.tar.zst failed to download
error: failed retrieving file 'linux-aarch64-rockchip-bsp6.1-joshua-git-6.1.43.r1266115.gc2daa578-1-aarch64.pkg.tar.zst' from github.com : The requested URL returned error: 404
warning: failed to retrieve some files
error: failed to commit transaction (failed to retrieve some files)
Errors occurred, no packages were upgraded.
```

I looked into the releases and there's no such file in there:

![image](https://github.com/7Ji/orangepi5-archlinuxarm/assets/8495707/a2fb5159-d0ac-4e05-ab62-192ecd3163d7)

I'm not sure how's it's built, so I cannot help more, unfortunately.

**Reply by 7Ji** [reply]
Sorry the release was desynced from my local repo, I've forced an update and everything should be back to normal

**Reply by squassina** [reply]
I was out of town, but was just able to check and it worked! Thanks!

**Comment by squassina** [comment]
Sorry, but I had to reinstall after messing with the system a little too much and now it's not accepting the 6.1 kernels, only the 5.10 that are in the original image. I'm following the steps in the README file but no luck.

Can you help me again? Here's my extlinux.conf:

```
MENU TITLE Select the kernel to boot

DEFAULT linux-aarch64-rockchip-bsp6.1-joshua-git

#linux-aarch64-rockchip-rk3588-bsp5.10-orangepi-git

#TIMEOUT 30

LABEL	linux-aarch64-rockchip-bsp6.1-joshua-git
	LINUX	/vmlinuz-linux-aarch64-rockchip-bsp6.1-joshua-git
	INITRD	/booster-linux-aarch64-rockchip-bsp6.1-joshua-git.img 
	#FDTDIR	/dtbs/linux-aarch64-rockchip-bsp6.1-joshua-git
	FDT	/dtbs/linux-aarch64-rockchip-bsp6.1-joshua-git/rockchip/rk3588-orangepi-5-plus.dtb
	APPEND	root=UUID=2757bd91-81be-45a1-9f7a-c27416524125 rw

LABEL	linux-aarch64-rockchip-rk3588-bsp5.10-orangepi-git
	LINUX	/vmlinuz-linux-aarch64-rockchip-rk3588-bsp5.10-orangepi-git
	INITRD	/booster-linux-aarch64-rockchip-rk3588-bsp5.10-orangepi-git.img
	#FDTDIR	/dtbs/linux-aarch64-rockchip-rk3588-bsp5.10-orangepi-git
	FDT	/dtbs/linux-aarch64-rockchip-rk3588-bsp5.10-orangepi-git/rockchip/rk3588-orangepi-5-plus.dtb
	APPEND	root=UUID=2757bd91-81be-45a1-9f7a-c27416524125 rw
``` 

Thanks!

**Reply by zoharxero** [reply]
I got it working installing [booster](https://wiki.archlinux.org/title/Booster) first, reinstalled the 6.1 kernel, then added the 6.1 kernel to extlinux.conf like your example, and removed the other kernel labels.

**Reply by squassina** [reply]
Commenting the other lines did the trick! Thank you!

**Comment by 7Ji** [comment]
Do not install kernels from official ALARM repo: they're not for opi, and they would bring in hard dependency of `mkinitcpio`, which is pretty much broken on ALARM.

Do not uninstall all kernels before installing 6.1 or my mainline 6.9: booster comes pre-installed as a provider for `initramfs` of kernel packages, removing all kernels with `pacman -Rcns` would make booster unneeded and removed along with the kernels. 

Do not remove existing extlinux.conf entries and kernels early, just add yours to the lowest of the list, and set yours as the new default. If the new kernel boots then it's fine; if it doesn't then u-boot would still try other kernels one-by-one. Only after successful reboots into new kernels, should you remove the old kernels.

Write your own extlinux.conf entries and don't rely on automatical tools, every one would have different booting configs.

**Reply by squassina** [reply]
Don't you worry! I'm installing from your repo, as I'm too lazy to start compiling myself.

I'm editing the extlinux.conf myself without other tools, too!

Thanks for your work!

**Comment by gelantoo** [comment]
Hello! Is there a next new kernel update? Armbian has 6.10 and it works! Wi-fi RTL8852BE, bluetooth, hardware acceleration in chromium and firefox, everything works.

**Reply by 7Ji** [reply]
https://github.com/orgs/7Ji-PKGBUILDs/repositories?q=linux-aarch64-

Focus is always to provide whatever upstream provides and do no local modification. -7ji is at 6.10, mainline; some others IIRC are at 6.10 with their midstream patches, and you can try them each. As for userspace and decoding, check chromium-mpp but try older 114. Most of PKGBUILDs beed to be built manually
